testng-core/src/test/java/test/timeout/github2672/TimeoutStacktraceTest.java

```diff
@@ -0,0 +1,41 @@
+package test.timeout.github2672;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+import java.util.Arrays;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+import org.testng.internal.thread.ThreadTimeoutException;
+
+@Listeners(TimeoutStacktraceTest.class)
+public class TimeoutStacktraceTest implements ITestListener {
```

Comment:
Don't mix test class and listener. Use inner class instead.
Add the test into the suite file.